library/Mockery/Container.php

```diff
@@ -233,7 +233,11 @@ public function mock(...$args)
         $mock->mockery_init($this, $config->getTargetObject(), $config->isInstanceMock());
 
         if (!empty($quickdefs)) {
-            $mock->shouldReceive($quickdefs)->byDefault();
+            if (\Mockery::getConfiguration()->getQuickDefinitionsApplicationMode() === \Mockery\Configuration::QUICK_DEFINITIONS_APPLICATION_MODE_DEFAULT_EXPECTATION) {
```

Comment:
Rather turn these if statements into questions on the configuration, like:

```
if (\Mockery::getConfiguration()->theQuestionMethodGoesHere()) {
$mock->...->byDefault();
} else {
$mock->...->atLeast()->once();
}
```

now as for the `theQuestionMethodGoesHere`... we should come up with a descriptive method name. `shouldQuickDefinitionSetupAStub()`, `shouldQuickDefinitionSetupAMock()`, `isQuickDefintionModeDefault()`, `shouldSetExpectationOnQuickDefinition()`... Something like this.

Reply:
Thing is the class constants are really long and make the code a bit unreadable, so we should come up with a better way of setting either of the two modes on the configuration, too.

Reply:
I see your point, before doing the changes, I would like to offer my point of view.
When I chose to use a 'setting' configuration over a boolean I did so, because I thought that a boolean would not model perfectly this behaviour, and it's not really extendable if we decide to add other behaviour in the future.

Indeed if we use both `shouldQuickDefinitionSetupAStub` and `shouldQuickDefinitionSetupAMock` we are not going to express that those are mutually exclusive from an API point of view. If we use just one (let' say just `shouldQuickDefinitionSetupAStub`), we're not descriptive enough about what (and why) should happen when it's false.

Still I understand how the longness and naming is not cool. What I would usually do in my code in those cases is to use an [enum-like](https://github.com/myclabs/php-enum) class that would look something like `QuickDefinitionMode` having as "values" `DEFAULT` and `MOCK_AT_LEAST_ONCE` at this point the getter and setter would look something like \Mockery\Configuration::setQuickDefinitionMode(QuickDefinitionMode::MOCK_AT_LEAST_ONCE())`. I think the lenght is fair enough and we would also be better because we could hint in the setter the use of this enum instead of a string (as of right now). But that would mean inserting that library in the code or use a surrogate where we just use the plain constants in a dedicated class (note how if you use the library when you use `QuickDefinitionMode::MOCK_AT_LEAST_ONCE()` you gey back a object of type `QuickDefinitionMode` not a string, while if we do it with our own class without magic we could just use the constant string)

So our options are:
- we're cool with no being expressive in the API, we can use 2 methods and document it and that's it.
- we introduce the enum for this setting.

Please let me know how you want to proceed and I'll do the changes.

thanks.